src/main/java/org/terasology/web/resources/OnlinePlayersResource.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.web.resources;
+
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.network.Client;
+import org.terasology.network.NetworkSystem;
+import org.terasology.network.events.ConnectedEvent;
+import org.terasology.network.events.DisconnectedEvent;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RegisterSystem
+public class OnlinePlayersResource extends ObservableReadableResource<List<String>> implements DefaultComponentSystem {
+
+    private NetworkSystem networkSystem;
+
+    OnlinePlayersResource(NetworkSystem networkSystem) {
```

Comment:
Initially I used constructor injection since it looked more test-friendly but I now (9be5efa commit) switched to the `@In`s to make it more coherent with the various other classes I saw in the engine repository, and after all it just requires a few more lines in the test cases, definitely not a big deal. Thanks for the suggestion.